src/utils.ts

```diff
@@ -320,13 +319,10 @@ export function getAPIFileType(filePath: string): string {
 export async function checkIfOnline() {
   // If using a proxy, return true since `isOnline` doesn't work.
   // @see https://github.com/googleapis/google-api-nodejs-client#using-a-proxy
-  if (process.env.HTTP_PROXY || process.env.HTTPS_PROXY) {
+  if (process.env.HTTP_PROXY || process.env.HTTPS_PROXY || (await isOnline())) {
     return true;
   }
-  if (!(await isOnline())) {
-    logError(null, ERROR.OFFLINE);
-    process.exit(1);
```

Comment:
I think `process.exit(1);` is needed.

Reply:
No. `logError()` never returns and end with its own `process.exit(1)`